liquidprompt

```diff
@@ -1550,6 +1556,78 @@ _lp_http_proxy_color() {
     lp_http_proxy_color="${LP_COLOR_PROXY}${LP_MARK_PROXY}${NO_COL}"
 }
 
+_lp_env_vars() { # [color_set, [color_unset]]
+    # Expects LP_ENV_VARS to be an array containing items of the form:
+    # "<ENV_VAR_NAME> <string if set>[ <string if unset>]"
+    # Strings may be `%s`, which will be replaced by the variable's actual content.
+
+    (( LP_ENABLE_ENV_VARS )) || return 2
+    [[ -n "${LP_ENV_VARS-}" && ${#LP_ENV_VARS} -gt 0 ]] || return 1
+
+    local color_set="${1-}"
+    local color_unset="${2-}"
+    local color_no="${NO_COL}"
+    if [[ -z "$color_set" && -z "$color_unset" ]]; then
+        color_no=""
+    fi
+
+    local var_rep var evar fmt_if_set fmt_if_unset
+
+    lp_env_vars=()
+    # For all user-defined setup.
+    for var_rep in ${LP_ENV_VARS[@]+"${LP_ENV_VARS[@]}"}; do
+        IFS=' ' read -r var fmt_if_set fmt_if_unset <<<"$var_rep"
+        # Variable name and format has to be set, but default is optional.
+        if [[ -n "${var}" && -n "${fmt_if_set}" ]]; then
+            # Expands the underlying variable name.
+            local var_is_set=
+            if (( _LP_SHELL_zsh )); then
+                # From https://www.shellcheck.net/wiki/SC2296 :
+                # "Some Zsh specific parameter expansions like ${(q)value} trigger this warning,
+                # but ShellCheck does not support Zsh."
+                # shellcheck disable=SC2296
+                var_is_set="${(P)var+__LP_IS_SET__}"
+            else
+                # NOTE: indirection expansion are deprecated starting at bash 4.3, should use nameref.
+                var_is_set="${!var+__LP_IS_SET__}"
+            fi
+
+            if [[ "$var_is_set" == "__LP_IS_SET__" ]]; then
```

Comment:
This seems overly complicated to me, why not
```suggestion
                var_is_set="${(P)var+x}"
            else
                # NOTE: indirection expansion are deprecated starting at bash 4.3, should use nameref.
                var_is_set="${!var+x}"
            fi

            if [[ -n $var_is_set ]]; then
```

Reply:
I find the syntax unreadable (as often with Bash), but at first I misunderstood (I believed we would have the content of var if it was set, hence using a unique placeholder). As least I would prefer to see `${!var+IS_SET}` which is more clear to me.

Reply:
Fair enough. The big one I wanted is the `-n` check instead of `==`, which you have changed.